### PR TITLE
gadgettracermanager: Make gRPC buffer size configurable.

### PR DESCRIPTION
**Pull request by eiffel-fl**
This commit adds flags and environment values to set gRPC buffer size rather than having it hardcoded.
The default value size is set to 16384 to improve horizontal scaling.

### CODE REVIEW

File: cmd/ig/daemon.go [hunk 1]
```diff
@@ -55,6 +56,13 @@ func newDaemonCommand(runtime runtime.Runtime) *cobra.Command {
 		"The socket to listen on for new requests. Can be a unix socket"+
 			" (unix:///path/to.socket) or a tcp socket (tcp://127.0.0.1:1234)")
 
+	daemonCmd.PersistentFlags().Uint64VarP(
+		&size,
```
Comment (flyth) [inline cmd/ig/daemon.go:60]: Maybe a more specific name like `eventBufferSize` or `eventBufferLen` is better here.

File: pkg/gadget-service/service.go [hunk 2]
```diff
@@ -56,16 +56,18 @@ type RunConfig struct {
...
 	api.UnimplementedGadgetManagerServer
-	listener net.Listener
-	runtime  runtime.Runtime
-	logger   logger.Logger
-	servers  map[*grpc.Server]struct{}
+	listener   net.Listener
+	runtime    runtime.Runtime
+	logger     logger.Logger
+	servers    map[*grpc.Server]struct{}
+	bufferSize uint64
```
Comment (flyth) [inline pkg/gadget-service/service.go:63]: Also here - maybe `eventBufferSize` or `eventBufferLen`.

**Review by flyth — COMMENTED** [review]
I think having this number configurable is a really good thing. At which point did it lead to issues? Can you please attach some data/benchmarks around it?

Also, this is about the buffer of events (in _number of events_, __not__ _number of bytes of event data_) that we're holding before marshaling them, so it's got nothing to do at that stage with gRPC, yet (gRPC uses buffers on its own, so let's not confuse by calling this gRPC buffer (naming again, I know... hehe) - even though it eventually will get sent over gRPC). So I think `eventBufferSize` or `eventBufferLen` would be a better name.

Just fyi, sending looks something like this:
```
Event -> __Event Buffer__ -> Marshaling -> gRPC -> TCP send buffer -> wire
```

The buffer is mainly used to throw away events if the client is slow in reading the events (and thus the tcp write on the server is blocking), and it especially handles bursts at the start of a gadget run - it also kicks in if marshaling is too slow to keep up (if the CPU is under heavy load for example). It would be interesting to know if increasing the TCP write buffer would also help.

**Review by burak-ok — APPROVED** [review]
From my side it looks good.

**Pull request merged**